src/machine/usb.go

```diff
@@ -402,17 +341,21 @@ func handleStandardSetup(setup USBSetup) bool {
 	}
 }
 
-// EnableHID enables HID. This function must be executed from the init().
-func EnableHID(callback func()) {
-	usbDescriptor = descriptorCDCHID
-	endPoints = []uint32{usb_ENDPOINT_TYPE_CONTROL,
-		(usb_ENDPOINT_TYPE_INTERRUPT | usbEndpointIn),
-		(usb_ENDPOINT_TYPE_BULK | usbEndpointOut),
-		(usb_ENDPOINT_TYPE_BULK | usbEndpointIn),
-		(usb_ENDPOINT_TYPE_INTERRUPT | usbEndpointIn)}
-
-	hidCallback = callback
+func EnableCDC(callback func(), callbackRx func([]byte), callbackSetup func(USBSetup) bool) {
+	usbDescriptorConfig |= usbDescriptorConfigCDC
+	endPoints[usb_CDC_ENDPOINT_ACM] = (usb_ENDPOINT_TYPE_INTERRUPT | usbEndpointIn)
+	endPoints[usb_CDC_ENDPOINT_OUT] = (usb_ENDPOINT_TYPE_BULK | usbEndpointOut)
+	endPoints[usb_CDC_ENDPOINT_IN] = (usb_ENDPOINT_TYPE_BULK | usbEndpointIn)
+	callbackUSBRx[usb_CDC_ENDPOINT_OUT] = callbackRx
+	callbackUSBTx[usb_CDC_ENDPOINT_IN] = callback
+	callbackUSBSetup[usb_CDC_ACM_INTERFACE] = callbackSetup // 0x02 (Communications and CDC Control)
+	callbackUSBSetup[usb_CDC_DATA_INTERFACE] = nil          // 0x0A (CDC-Data)
 }
 
-// hidCallback is a variable that holds the callback when using HID.
-var hidCallback func()
+// EnableHID enables HID. This function must be executed from the init().
+func EnableHID(callback func(), callbackRx func([]byte), callbackSetup func(USBSetup) bool) {
```

Comment:
Rename params to `handler`, `rxHandler`, `setupHandler`?

Reply:
Fixed.